kpm installs linked to a raw coffee script fail to install

`/kpm install xkcd` failed to install with the following message

```
fatal: repository 'https://raw.githubusercontent.com/github/hubot-scripts/master/src/scripts/xkcd.coffee/ ' not found
Cloning into '/tmp/tmp-4472Y2VPkienKvC8'...
fatal: repository 'https://raw.githubusercontent.com/github/hubot-scripts/master/src/scripts/xkcd.coffee/ ' not found
```

But was successfully installed when manually linking the script as

`/kpm install https://raw.githubusercontent.com/github/hubot-scripts/master/src/scripts/xkcd.coffee`

Possible cause is the `/` at the end of the link added by kpm

Tested installs and uninstalls via `kpm` table and with links. Works fine as expected but one link did stuff it up

`/kpm install https://raw.githubusercontent.com/hubot-scripts/hubot-sentimental/master/src/sentimental.coffee`

with the following message

```
Loading module 'sentimental'... 
Error: Cannot find module 'Sentimental'
  at Function.Module._resolveFilename (module.js:339:15)
  at Function.Module._load (module.js:290:25)
  at Module.require (module.js:367:17)
  at require (internal/module.js:16:19)
  at Object.<anonymous> (/root/Dev/kassy_test/modules/kpm_sentimental/sentimental.coffee:21:11)
  at Object.<anonymous> (/root/Dev/kassy_test/modules/kpm_sentimental/sentimental.coffee:21:1)
  at Module._compile (module.js:413:34)
  at Object.loadFile (/root/Dev/kassy_test/node_modules/coffee-script/lib/coffee-script/register.js:16:19)
  at Module.load (/root/Dev/kassy_test/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:314:12)
  at Module.require (module.js:367:17)
  at require (internal/module.js:16:19)
  at Object.exports.requireOrInstall (/root/Dev/kassy_test/core/install.js:42:12)
  at func (/root/Dev/kassy_test/core/require.js:20:21)
  at Function.func.once (/root/Dev/kassy_test/core/require.js:57:19)
  at Object.exports.loadModule (/root/Dev/kassy_test/core/modules/hubotModule.js:108:25)
  at Object.exports.loadModule (/root/Dev/kassy_test/core/modules/modules.js:37:45)
  at /root/Dev/kassy_test/core/core_modules/kpm.js:134:52
  at doneOne (/root/Dev/kassy_test/node_modules/fs-extra/lib/copy/ncp.js:237:40)
  at /root/Dev/kassy_test/node_modules/fs-extra/lib/copy/ncp.js:122:11
  at FSReqWrap.oncomplete (fs.js:82:15)
Loading Failed
Could not load module 'sentimental'. Does it have a syntax error?
Module "sentimental" could not be loaded.
```

--- a/core/core_modules/kpm.js
+++ b/core/core_modules/kpm.js
@@ -6,6 +6,7 @@ var git = require.once('../git.js'),
     sanitize = require.safe('sanitize-filename'),
     request = require('request'),
     urll = require('url'),
+    deasync = require('deasync'),
     moduleCache = null,
     moduleTable = {
         lastUpdated: null,
@@ -214,39 +215,39 @@ var git = require.once('../git.js'),
             return callback(url);
         }
 
-        request.get('https://raw.githubusercontent.com/wiki/mrkno/Kassy/KPM-Table.md', function (error, response) {
-            if (response.statusCode === 200 && response.body) {
-                var b = response.body;
-                if (b && b.length > 0) {
-                    var spl = b.split('\n'),
-                        shouldParse = false,
-                        foundModules = {};
-                    for (var i = 0; i < spl.length; i++) {
-                        if (!spl[i].startsWith('|')) {
-                            continue;
-                        }
-
-                        var items = spl[i].split('|');
-                        if (items.length !== 4) {
-                            continue;
-                        }
-                        if (!shouldParse) {
-                            if (items[1] === '---' && items[2] === '---') {
-                                shouldParse = true;
-                            }
-                            continue;
-                        }
-                        foundModules[items[1]] = items[2];
+        var sreq = deasync(request.get);
+        var response = sreq('https://raw.githubusercontent.com/wiki/mrkno/Kassy/KPM-Table.md');
+        if (response.statusCode === 200 && response.body) {
+            var b = response.body;
+            if (b && b.length > 0) {
+                var spl = b.split('\n'),
+                    shouldParse = false,
+                    foundModules = {};
+                for (var i = 0; i < spl.length; i++) {
+                    if (!spl[i].startsWith('|')) {
+                        continue;
                     }
-                    moduleTable.modules = foundModules;
-                    moduleTable.lastUpdated = new Date();
+
+                    var items = spl[i].split('|');
+                    if (items.length !== 4) {
+                        continue;
+                    }
+                    if (!shouldParse) {
+                        if (items[1] === '---' && items[2] === '---') {
+                            shouldParse = true;
+                        }
+                        continue;
+                    }
+                    foundModules[items[1]] = items[2];
                 }
-                callback(url);
+                moduleTable.modules = foundModules;
+                moduleTable.lastUpdated = new Date();
             }
-            else {
-                callback(url, 'Could not update the list of KPM entries. Module entries may not be up to date.');
-            }
-        });
+            callback(url);
+        }
+        else {
+            callback(url, 'Could not update the list of KPM entries. Module entries may not be up to date.');
+        }
     },
 
     opts = {
@@ -287,29 +288,30 @@ var git = require.once('../git.js'),
 
                 for (var i = 0; i < args.length; i++) {
                     var url = args[i];
+                    var spl = url.split('/');
+                    if (spl.length === 1) {
+                        refreshModuleTable(url, function(u, err) {
+                                if (err || !moduleTable.modules[u]) {
+                                    return;
+                                }
+                                url = moduleTable.modules[u];
+                            }.bind(this));
+                    }
+                    else if (!url.startsWith('ssh') && !url.startsWith('http')) {
+                        if (spl.length === 2) {
+                            url = 'https://github.com/' + url.trim();
+                        }
+                        else {
+                            api.sendMessage('Invalid KPM module provided "' + url + '". Skipping...', event.thread_id);
+                            continue;
+                        }
+                    }
+
                     if (url.startsWith('ssh') || url.endsWith('.git')) {
                         gitInstall.call(this, url, api, event);
-                        continue;
                     }
                     else if (url.startsWith('http') && (url.endsWith('.coffee') || url.endsWith('.js'))) {
                         scriptInstall.call(this, url, api, event);
-                        continue;
-                    }
-
-                    var spl = url.split('/');
-                    if (spl.length === 1) {
-                        refreshModuleTable(url, function(url, err) {
-                            if (err || !moduleTable.modules[url]) {
-                                api.sendMessage('Invalid KPM table reference provided "' + url + '". Skipping...', event.thread_id);
-                                return;
-                            }
-                            url = moduleTable.modules[url];
-                            gitInstall.call(this, url, api, event);
-                        }.bind(this));
-                    }
-                    else if (spl.length === 2) {
-                        url = 'https://github.com/' + url.trim();
-                        gitInstall.call(this, url, api, event);
                     }
                     else {
                         api.sendMessage('Invalid KPM module provided "' + url + '". Skipping...', event.thread_id);

--- a/core/core_modules/kpm.js
+++ b/core/core_modules/kpm.js
@@ -291,11 +291,11 @@ var git = require.once('../git.js'),
                     var spl = url.split('/');
                     if (spl.length === 1) {
                         refreshModuleTable(url, function(u, err) {
-                                if (err || !moduleTable.modules[u]) {
-                                    return;
-                                }
-                                url = moduleTable.modules[u];
-                            }.bind(this));
+                            if (err || !moduleTable.modules[u]) {
+                                return;
+                            }
+                            url = moduleTable.modules[u];
+                        }.bind(this));
                     }
                     else if (!url.startsWith('ssh') && !url.startsWith('http')) {
                         if (spl.length === 2) {

--- a/core/install.js
+++ b/core/install.js
@@ -12,11 +12,25 @@
 var npm = require('npm'),
     deasync = require('deasync'),
     fs = require('fs'),
-    load = deasync(npm.load);
+    load = deasync(npm.load),
+    csHasLoaded = false;
 
 load({loglevel: 'silent'});
 var inst = deasync(npm.commands.install),
-upd = deasync(npm.commands.update),
+    upd = deasync(npm.commands.update),
+
+// inject require modifications into all coffeescript code because babel wont
+coffeescriptRequireInjector = function () {
+    if (!csHasLoaded && global.coffeescriptLoaded) {
+        var cs = require("coffee-script"),
+            orig = cs._compileFile;
+        cs._compileFile = function () {
+            var res = orig.apply(this, arguments);
+            return require('./require.js').injectionString + res;
+        };
+        csHasLoaded = true;
+    }
+},
 
 install = function(name) {
     console.info('Installing "' + name + '" from npm.');
@@ -26,6 +40,7 @@ install = function(name) {
 
 exports.requireOrInstall = function(req, name) {
     try {
+        coffeescriptRequireInjector();
         return req(name);
     }
     catch (e) {

--- a/core/install.js
+++ b/core/install.js
@@ -19,24 +19,24 @@ load({loglevel: 'silent'});
 var inst = deasync(npm.commands.install),
     upd = deasync(npm.commands.update),
 
-// inject require modifications into all coffeescript code because babel wont
-coffeescriptRequireInjector = function () {
-    if (!csHasLoaded && global.coffeescriptLoaded) {
-        var cs = require("coffee-script"),
-            orig = cs._compileFile;
-        cs._compileFile = function () {
-            var res = orig.apply(this, arguments);
-            return require('./require.js').injectionString + res;
-        };
-        csHasLoaded = true;
-    }
-},
+    // inject require modifications into all coffeescript code because babel wont
+    coffeescriptRequireInjector = function () {
+        if (!csHasLoaded && global.coffeescriptLoaded) {
+            var cs = require('coffee-script'),
+                orig = cs._compileFile;
+            cs._compileFile = function () {
+                var res = orig.apply(this, arguments);
+                return require('./require.js').injectionString + res;
+            };
+            csHasLoaded = true;
+        }
+    },
 
-install = function(name) {
-    console.info('Installing "' + name + '" from npm.');
-    inst([name]);
-    console.info('Installation complete.');
-};
+    install = function(name) {
+        console.info('Installing "' + name + '" from npm.');
+        inst([name]);
+        console.info('Installation complete.');
+    };
 
 exports.requireOrInstall = function(req, name) {
     try {

--- a/core/modules/hubotModule.js
+++ b/core/modules/hubotModule.js
@@ -16,8 +16,7 @@ var fs = require('fs'),
     modulesDir = 'modules',
     descriptor = 'hubot.json',
     pkg = 'package.json',
-    Robot = require.once('./hubot/robot.js'),
-    coffeescriptLoaded = false;
+    Robot = require.once('./hubot/robot.js');
 
 var verifyModuleDescriptior = function (hj, disabled) {
     if (!hj.name || !hj.startup || !hj.version) {
@@ -99,9 +98,9 @@ exports.listModules = function (disabled) {
 };
 
 exports.loadModule = function (module) {
-    if (!coffeescriptLoaded && module.startup.endsWith('.coffee')) {
+    if (!global.coffeescriptLoaded && module.startup.endsWith('.coffee')) {
         require('coffee-script').register();
-        coffeescriptLoaded = true;
+        global.coffeescriptLoaded = true;
     }
 
     try {

--- a/core/require.js
+++ b/core/require.js
@@ -66,12 +66,12 @@ module.exports = function () {
     return {
         visitor: {
             Program(path) {
-                path.unshiftContainer("body", babylon.parse(module.exports.injectionString).program.body[0]);
+                path.unshiftContainer('body', babylon.parse(module.exports.injectionString).program.body[0]);
             }
         }
     };
 };
 
-module.exports.injectionString = "require = (global || GLOBAL).requireHook(require);";
+module.exports.injectionString = 'require = (global || GLOBAL).requireHook(require);';
 
 exports.default = module.exports;

--- a/core/require.js
+++ b/core/require.js
@@ -66,9 +66,12 @@ module.exports = function () {
     return {
         visitor: {
             Program(path) {
-                path.unshiftContainer('body', babylon.parse('require = (global || GLOBAL).requireHook(require);').program.body[0]);
+                path.unshiftContainer("body", babylon.parse(module.exports.injectionString).program.body[0]);
             }
         }
     };
 };
+
+module.exports.injectionString = "require = (global || GLOBAL).requireHook(require);";
+
 exports.default = module.exports;

